htmresearch/algorithms/column_pooler.py

```diff
@@ -403,26 +410,89 @@ def _learnProximal(self,
     for cell in activeCells:
       cellPermanencesDense = proximalPermanences.getRow(cell)
       cellNonZeroIndices, _ = proximalPermanences.rowNonZeros(cell)
-      cellNonZeroIndices = set(cellNonZeroIndices)
-
-      # Find the synapses that should be reinforced, punished, and grown.
-      reinforce = list(activeInputs & cellNonZeroIndices)
-      punish = list(cellNonZeroIndices - activeInputs)
-      growthCandidates = activeInputs - cellNonZeroIndices
-      newSynapseCount = min(len(growthCandidates), maxNewSynapseCount)
-      grow = _sample(growthCandidates, newSynapseCount, self._random)
-
-      # Make the changes.
-      cellPermanencesDense[punish] -= synPermProximalDec
-      cellPermanencesDense[reinforce] += synPermProximalInc
-      cellPermanencesDense[grow] = initialPermanence
-
-      # Update proximalPermanences and proximalConnections.
+      cellNonZeroIndices = list(cellNonZeroIndices)
```

Comment:
@ywcui1990 Looks like you are reverting some optimization changes here that were recently made by @mrcslws . Can you first merge with master in your branch and then update the PR?